src/MouseAdapter.js

```diff
@@ -309,20 +383,196 @@ define(function(require) {
+         * observable if desired in the future.
+         * @param {object} event
+         * @private
+         */
+        _dispatchUserReScroll: function(event) {
             this._dispatchMouseWheelEnd(event);
+            this._dispatchMouseWheelStart(event);
+        },
+
+        /**
+         * Detect when the user performs a new scroll gesture.
+         * @param {object} normalizedEvent
+         * @private
+         */
+        _detectUserScroll: function(normalizedEvent) {
+
+            /*
+             * This is a helper function which takes in a set of consecutive linear data 
+             * and tries to determine whether it is increasing or decreasing.  It can
+             * operate on an entire array, or only a specified section of it. When given a
+             * starting point and a length that extends outside the boundaries of the array,
+             * the function assumes the data wraps around to the beginning of the array.
+             * @param {Array.<number>} array The array of data to perform analysis on.
+             * @param {number} arrayMaxLength [optional] The max length of the array.  If
+             * not given, assumed to be array.length
+             * @param {number} startAt [optional] The index in the array where the target 
+             * data begins.  If not specified, assumes this index is 0.
+             * @param {number} length [optional] The number of data points to examine.  
+             * If not specified, assumed to be the entire array length.
+             * @return {number} Returns:
+                    1 if data appears to be increasing.
+                   -1 if data appears to be decreasing.
+                    0 if there is no discernable trend.
+             */
+            var calculateDeltaDirection = function(array,startAt,length,arrayMaxLength) {
+                arrayMaxLength = arrayMaxLength || array.length;
+                startAt = startAt || 0;
+                length = length || arrayMaxLength;
+                var total = 0;
+                var offsetIdx1,offsetIdx2;
+                for (var idx1 = 0; idx1 < length; idx1++) {
+                    for (var idx2 = idx1+1; idx2 < length; idx2++) {
+                        offsetIdx1 = (idx1+startAt) % arrayMaxLength;
+                        offsetIdx2 = (idx2+startAt) % arrayMaxLength;
+                        if (array[offsetIdx1] < array[offsetIdx2]) {
+                            total += (1/(idx2-idx1));
+                        }
+                        if (array[offsetIdx1] > array[offsetIdx2]) {
+                            total += (-1/(idx2-idx1));
+                        }
+                    }
+                }
+                if (total >= 1) {
+                    return 1; // Increasing trend
+                }
+                if (total <= -1) {
+                    return -1; // Decreasing trend
+                }
+                return 0; // Stable
```

Comment:
Yeah, it took me a long time to come up with this function, as i was focused on trying to find a function that **could** do direction and magnitude of trend, because that would have been awesome, but it proved to be completely futile as soon as noise gets introduced into the data.

`trend([10, 9, 8, 7, 100, 6, 5, 4, 3, 2, 1]` is clearly decreasing overall when plotted on a graph, but for a function which cares about magnitude, all it can see is a really large increase which outweighs the really small (though much more consistent) downward trend.  As soon as i could come up with logic to counter this edge case, i had also discovered a new edge case which destroys it.

After tons of research on the internet, general consensus was that data analysis is hard, and there is no "good" way of doing this which is mathematically valid in the general case.  So i abandoned magnitude entirely, and am actually very happy with this function.  It seems to be very accurate.